cloudbridge/cloud/base/services.py

```diff
@@ -62,6 +62,18 @@ class BaseSecurityService(SecurityService, BaseCloudService):
     def __init__(self, provider):
         super(BaseSecurityService, self).__init__(provider)
 
+    @property
+    def key_pairs(self):
+        return self._key_pairs
```

Comment:
this probably shouldn't be in the base class because the base class doesn't have a property called _key_pairs. Subclasses should probably redefine this despite being redundant because someone implementing a new provider may not notice that this is actually unimplemented.

Reply:
Ok I will move them back!